htdocs/resource/class/dolresource.class.php

```diff
@@ -460,24 +490,24 @@ public function fetchElementResource($id)
 	 *
 	 * @param	int		$rowid			Id of resource line to delete
 	 * @param	int		$notrigger		Disable all triggers
-	 * @return	int						Return integer if OK: >0, if KO: <0
+	 * @return	int						if OK: >0 || if KO: <0
 	 */
-	public function delete($rowid, $notrigger = 0)
+	public function delete(int $rowid, int $notrigger = 0): int
 	{
-		global $user, $langs, $conf;
+		global $user, $conf;
 		require_once DOL_DOCUMENT_ROOT.'/core/lib/files.lib.php';
 
 		$error = 0;
 
 		$this->db->begin();
 
 		$sql = "DELETE FROM ".MAIN_DB_PREFIX.$this->table_element;
-		$sql .= " WHERE rowid = ".((int) $rowid);
+		$sql .= " WHERE rowid = ".($rowid);
```

Comment:
We must force the int when forging sql request. Even if we are sure that it is an int, so it will avoid false positive in our tool to dectect sql injection that analyze the sql syntax.

Reply:
I think for phpstan it's ok. Thanks for explanation about sql injection. I will adjust code.